app.js

```diff
@@ -48,10 +37,38 @@ angular.module('PathOfDamage', [])
         }
       }
     }
-    $scope.updateHits(true);
-    if (!skipStringify) {
-      $scope.stringifyUrlData();
+    if (!skipUpdates) {
+      $scope.updateHits();
+    }
+  };
+
+  $scope.updateHits = function (skipSerialization) {
```

Comment:
I would prefer a more descriptive name, like UpdateDamageCalculation or something. 'hits' is used to mean not that in most of the code.